src/fields/TextField.cpp

```diff
@@ -1,9 +1,7 @@
 #include "TextField.h"
 
-TextField::TextField(string fieldName, string defaultValue, bool isPrimaryKey, bool isUnique, bool isNullable):
-    Field(fieldName, TEXT, isPrimaryKey, isUnique, isNullable) {
-    _fieldValue = defaultValue;
-}
+TextField::TextField(string &fieldName, string defaultValue, bool isPrimaryKey, bool isUnique, bool isNullable):

```

Comment:
The problem here is that in `TextField.h`/`CharField.h`, this has a value `defaultValue=""`.
So It throws an error if you try to pass by reference. I still need to see if I can find a way around this...